docs/proposals/authentication-filter.md

```diff
@@ -22,3 +22,1110 @@ This new filter should eventually expose all forms of authentication available t
+      stripAuthorization: true # Optionally remove client Authorization header before proxy_pass
+```
+
+### Caching configuration
+
+Users may also choose to change the caching configuration set by `proxy_cache_path`.
+This can be made available in the `cache` configuration under `jwt.remote.cache`
+
+```yaml
+kind: AuthenticationFilter
+metadata:
+  name: jwt-auth
+spec:
+  type: JWT
+  jwt:
+    realm: "Restricted"
+    mode: Remote
+    remote:
+      url: https://issuer.example.com/.well-known/jwks.json
+      cache:
+        path: /var/cache/nginx/jwks # required when cache is set
+        levels: "1:2"               # optional; defaults to "1:2"
+        keysZoneName: jwks_jwtauth  # optional; controller can default to a derived name
+        keysZoneSize: 10m           # required; size for keys_zone
+        maxSize: 50m                # optional; limit total cache size
+        inactive: 10m               # optional; inactivity TTL before eviction
+        useTempPath: false          # optional; sets use_temp_path
+```
+
+### Attachment
+
+Filters must be attached to a HTTPRoute at the `rules.matches` level.
+This means that a single `AuthenticationFilter` may be attached mutliple times to a single HTTPRoute.
+
+#### Basic example
+
+This example shows a single HTTPRoute, with a single `filter` defined in a `rule`
+
+![reference-1](/docs/images/authentication-filter/reference-1.png)
+
+### Status
+
+#### Referencing multiple AuthenticationFilter resources in a single rule
+
+Only a single `AuthenticationFilter` may be referened in a single rule.
+
+The `Status` the HTTPRoute/GRPCRoute in this scenario should be set to `Invalid`, and the resource should be `Rejected`.
+
+This behaviour falls in line with the expected behaviour of filters in the Gateway API, which generally allows only one type of a specific filter (authentication, rewriting, etc.) within a rule.
+
+Below is an example of an **invalid** HTTPRoute that references multiple `AuthenticationFilter` resources in a single rule
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: invalid-httproute
+spec:
+  parentRefs:
+  - name: gateway
+  hostnames:
+  - api.example.com
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /api
+    filters:
+    - type: ExtensionRef
+      extensionRef:
+        group: gateway.nginx.org
+        kind: AuthenticationFilter
+        name: basic-auth
+    - type: ExtensionRef
+      extensionRef:
+        group: gateway.nginx.org
+        kind: AuthenticationFilter
+        name: jwt-auth
+    backendRefs:
+    - name: backend
+      port: 80
+```
+
+#### Attaching a JWT AuthenticationFilter to a route when using NGINX OSS
+
+If a user attempts to attach a JWT tpye AuthenticationFilter while using NGINX OSS, the rule referncing the filter will be `Rejected`.
+
+This can appear as `UnresolvedRef` to inform the user that the rule has been `Rejected`.
```

Comment:
Is `Invalid` an option? UnresolvedRef generally means "we couldn't find this resource".

Reply:
@sjberman @salonichf5 let me know if you still think using `UnresolvedRef` as the status is correct here.
I'm still quite new to how these kind of conditions work for the GatewayAPI so I'm open to feedback and opinions on this one. Thanks!

Reply:
Okay, so in our case we need declare the rule as "invalid" and we will automatically add `PartiallyInvalid` by default today?

Also, when a route is `PartiallyInvalid` what is the outcome?
For example, if I have a valid route with a valid `AuthenticationFilter`, and I try to add another `AuthenticationFilter`, which would be invalid. Do we keep the pervious working state or does the route rule (`location`) get removed from NGINX?